### DISCUSSION TITLE
Tagesertragsdaten speichern?

### DISCUSSION THREAD

**Discussion by NintendoTito** [tbnobody/OpenDTU#954] [Ideas]
Ich bin kein Programmierer und kann die Machbarkeit nicht einschätzen, daher bitte nur als Anregung und nicht falsch verstehen. Ich schätze die Arbeit die hier die Entwickler leisten sehr!

Ich weiß, dass man per MQTT von der OpenDTU Daten senden und in Datenbanken archivieren und auswerten kann. Habe ich so halbwegs geschafft, ist aber für den "Normalo" doch schon relativ komplex. Daher meine Idee:

Wäre es möglich, dass die OpenDTU direkt auf dem ESP32 die täglichen Erträge speichert? Das erforderliche Datenvolumen bzw. der Speicherplatz sollte doch minimal sein (365 Werte pro Jahr)? Man hätte dann annähernd eine "All in One" Lösung und könnte über die Weboberfläche der OpenDTU dann auch diese Tageserträge für verschiedene Zeiträume (Wochen, Monate) grafisch auswerten.

Dann könnte die OpenDTU annähernd alles wofür ansonsten ein Shelly notwendig wäre, also einfache Auswertungen der erzeugten Mengen für den "Otto-Normal-Nutzer".

**Comment by jblond4711** [comment]
Hi,

diesem Wunsch schließe ich mich vollumfänglich an.
Falls der Platz auf dem Esp32 nicht reicht gäbe es irgendwie die Möglichkeit die Tagesdaten zu verschicken?
Gäbe es denn die Möglichkeit einen Micro SD Kartenleser an den Esp32 zu „basteln“?

**Reply by NintendoTito** [reply]
"Verschicken" kann man die Tagesdaten ja per MQTT. Dann muss man halt "dahinter" noch viel basteln,also zb Raspberry mit Node-Red, InfluxDB oder iobroker etc. Halt schon nicht so "mal eben" gemacht...

**Reply by Nobster18** [reply]
Ich stand vor demselben Problem, wollte mir die Daten gerne auch mal über einen längeren Zeitraum ansehen, habe da aber so auf Anhieb nicht gefunden. Da ich hier noch einen "älteren" Raspberry 3B rumliegen hatte, habe ich mich da mal auf die Suche gemacht und das Projekt "Solaranzeige" gefunden.
Nach ein wenig Bastelei kommt genau das raus, was ich gerne haben wollte. Eine Datenbank, die die Werte über einen längeren Zeitraum speichert und diese mittels Grafana visualisiert.
Screenshots usw. findet man auf der Homepage. Viel Spaß beim Nachbauen - ist auch nicht so schrecklich kompliziert. Oka, ich arbeite in der IT und mache schon länger mit Raspberries rum, daher bin ich sicherlich auch nicht der Maßstab. Aber schaut Euch das mal an, ich denke, dass ist das was Ihr möchtet.
https://solaranzeige.de/phpBB3/solaranzeige.php

**Comment by NintendoTito** [comment]
Ich habe erst eben festgestellt, dass diese Discussion in die gleiche Richtung geht:
https://github.com/tbnobody/OpenDTU/discussions/176

Und ja, Hardware für ne SD Karte scheint es zu geben. Allerdings fände ich die "einfache" Lösung, also lediglich die Tagesertragsdaten zu speichern sinnvoller.

**Comment by jblond4711** [comment]
Hi,

mit verschicken meinte ich nicht MQTT sondern die Tagesdaten Datei per Push, eMail, iCloud Drive oder was es sonst noch so gäbe.

**Comment by DPO99** [comment]
Die originale DTU (selbst) speichert m.E. keine Daten (das macht die HM Cloud). Ich glaube nicht, dass man den eh schon sehr begrenzen Speicher mit so etwas volladen möchte. Man muss das dann ja regelmäßig löschen und was genau ist denn das Ziel? Wenn man eine Art Logging oder History aufbauen will, dann ist die DTU dafür eigentlich nicht gemacht. Dafür gibt es ja div. Schnittstellen dahin. Das ganze geht ja noch weiter...wenn man die Daten speichern will, braucht man auch eine Option diese anzuzeigen..am besten mit Grafik? Wieder mehr Speicher und Ressourcen fresser....dann ist die DTU für bis zu 10WR gebaut...bei 10 HM kommt dann schon einiges mehr an Daten zusammen.....Vermutlich ist dann die Verfolgung der Idee 176 die beste Idee.

**Comment by jblond4711** [comment]
Hi,

persönlich bräuchte ich nur den Tageswert, meinetwegen als reinen Textwert, um ihn in eine Exel Tabelle einzutragen.
Solange ich abends zu Hause bin um den Wert abzulesen ist das kein Problem. Aber man ist ja nicht jeden Abend zu Hause oder vergisst es wegen anderer Sachen auch mal.

**Comment by bennoulli** [comment]
Hi, I would also suggest the idea 
- to keep record in the openDTU the daily yield ("Tagesertrag") at the end of the day, e.g. for up to 365 days
- and the hourly values of the daily yield, maybe for up to 30 days
- to store this data for a convenient period on the openDTU, and provide these naked values in the web interface (text output).

It could be a nice solution for those who actually like to record the inverter history and build their own statistics - without MQTT and other permanently active equipment.

and thanks to all the active developers and contributors, great respect for those who dedicate their talent to this great project!

**Comment by LeChrisDE** [comment]
Ich selbst nutze die Prometheus-Schnittstelle und stelle die Inhalte direkt in meinem Grafana-Dashboard dar. 
Workflow: OpenDTU -> Prometheus (via metrics-URI: siehe https://github.com/tbnobody/OpenDTU/blob/master/docs/Web-API.md) -> Grafana
Dafür braucht es "nur" eine Prometheus und Grafana - Installation. Somit ist dieser deutlich kürzer als der Weg über MQTT -> MQTT-Broker (ioBroker, Mosquitto) -> Datenbank (Influx o.ä.) -> Grafana.

Das Problem: Auch dafür braucht es ein wenig IT-Nähe. Vielen Menschen (wie z.B. jblond4711) würde es sicherlich genügen, wenn die OpenDTU eine Funktion implementiert bekommen könnte, um (z.B. zu einer fixen Uhrzeit) die Werte des Tages (zur Not auch einfach "roh") an eine zu konfiurierende E-Mail - Adresse zu senden.

Dies wäre (m)ein Vorschlag. Gleichzeitig möchte ich den EntwicklerInnen für Ihre wahnsinnig tolle Arbeit am Projekt danken!

**Comment by jblond4711** [comment]
Hi,

ein verschicken per E-Mail oder generell verschicken braucht es meinetwegen ja nicht, war nur wegen meiner Faulheit.
Könnte man nicht einfach eine klitzekleine Testdatei mit dem Tageswert speichern? Diese Datei muss ja auch nicht Monatelang mitgeschleppt werden.

**Comment by kinichahau2023** [comment]
To Keeper daily values please check out:
https://github.com/RaBa64/OpenDTU-Database/releases/

**Reply by bennoulli** [reply]
I have tested this fork of RaBa, and I like this one a lot ... it shows an hourly yield graph for the last 24 hours and a 365 days history in the web interface. Plus, you can download hourly yield as a CSV like file on the web interface.
Still it could deserve some polishing, but I would encourage to merge its main work (keep hourly yield over 365 days on openDTU and allow their download in a file) to this project.

**Reply by MetaChuh** [reply]
@bennoulli 
very old thread, but thanks for pointing this out 👍

@RaBa64
du könntest deine [<kbd>README.md</kbd>](https://github.com/RaBa64/OpenDTU-Database/blob/master/README.md) mit allen zusatz features ergänzen, damit dich mehr user finden können.

... jedoch ... wenn man sich deinen erfrischend schönen und leeren issue tracker ansieht, ist es evtl. besser nur von einer kleinen gruppe gefunden zu werden (damit alles so bleibt) 😉

greetings,
metachuh

**Reply by francwalter** [reply]
Stimmt, hab ich auch eher zufällig entdeckt, danke @RaBa64, ich habe die FW gleich aufgespielt.

**Comment by Tiese** [comment]
Ich hätte da was: [OpenDTU-Client V2](https://opendtu-client-v2.kurze.email). &#128515;
Der verschickt nicht nur täglich eine Email (@jblond4711: was du ja schon mal versucht hast in OpenDTU-OnBattery einzuprogrammieren) mit den Tagesdaten, sondern kann auch noch einiges mehr. &#129321;
Zudem ist er wirklich ganz easy einzurichten!

Schöne Grüße
Tiese
